proto/v1/structure/state.proto

```diff
@@ -0,0 +1,455 @@
+  StateStructureVersion structure_version = 1;
+
+  SubtitledHtml content = 2;
+  InteractionInstance interaction = 3;
+  RecordedVoiceovers recorded_voiceovers = 4;
+  WrittenTranslations written_translations = 5;
+}
+
+// Structure corresponds to StateStructureVersion.
+message InteractionInstance {
+  oneof interaction_type {
+    ContinueInstance continue = 1;
+    FractionInputInstance fraction_input = 2;
+    ItemSelectionInputInstance item_selection_input = 3;
+    MultipleChoiceInputInstance multiple_choice_input = 4;
+    NumericInputInstance numeric_input = 5;
+    TextInputInstance text_input = 6;
+    DragAndDropSortInputInstance drag_and_drop_sort_input = 7;
+    ImageClickInputInstance image_click_input = 8;
+    RatioExpressionInputInstance ratio_expression_input = 9;
+    EndExplorationInstance end_exploration = 10;
+  }
+}
+
+// Answer type: none (N/A since no answers are submitted for this interaction).
+// Structure corresponds to StateStructureVersion.
+message ContinueInstance {
+  CustomizationArgs customization_args = 1;
+  Outcome default_outcome = 2;
+
+  // Continue interactions cannot have custom answer groups, and do not support
+  // solutions. Users cannot receive hints since only one answer is possible
+  // (clicking the button).
+
+  message CustomizationArgs {
+    // Note that while the continue interaction has a default button text
+    // defined, it's not used in the app since the default string needs to come
+    // through the app's strings so that it can be translated.
+    SubtitledHtml button_text = 1;
+  }
+}
+
+// Answer type: Fraction.
+// Structure corresponds to StateStructureVersion.
+message FractionInputInstance {
+  CustomizationArgs customization_args = 1;
+  repeated AnswerGroup answer_groups = 2;
+  Outcome default_outcome = 3;
+  repeated Hint hints = 4;
+  Solution solution = 5;
+
+  message CustomizationArgs {
+    bool requires_simplest_form = 1;
+    bool allow_improper_fractions = 2;
+    bool allow_nonzero_integer_part = 3;
+    SubtitledHtml custom_placeholder = 4;
+  }
+  message Solution {
+    BaseSolution base_solution = 1;
+    Fraction correct_answer = 2;
+  }
+  message AnswerGroup {
+    BaseAnswerGroup base_answer_group = 1;
+    repeated RuleSpec rule_specs = 2;
+  }
+  message RuleSpec {
+    oneof rule_type {
+      IsExactlyEqualToSpec is_exactly_equal_to = 1;
+      IsEquivalentToSpec is_equivalent_to = 2;
+      IsEquivalentToAndInSimplestFormSpec is_equivalent_to_and_in_simplest_form = 3;
+      IsLessThanSpec is_less_than = 4;
+      IsGreaterThanSpec is_greater_than = 5;
+      HasNumeratorEqualToSpec has_numerator_equal_to = 6;
+      HasDenominatorEqualToSpec has_denominator_equal_to = 7;
+      HasIntegerPartEqualToSpec has_integer_part_equal_to = 8;
+      HasNoFractionalPartSpec has_no_fractional_part = 9;
+      HasFractionalPartExactlyEqualToSpec has_fractional_part_exactly_equal_to = 10;
+    }
+    message IsExactlyEqualToSpec {
+      Fraction input = 1;
```

Comment:
Fraction message not present

Reply:
Where do you see this? Bazel seems to find it fine per the imports at the top of the file.

Reply:
I missed it earlier, found the Fraction message inside `object.proto`

Reply:
Got it. Does that mean this seems correct to you?

Reply:
Yes, resolving the conversation as I have nothing more to ask here.